### PR TITLE
Add YASMIN to rosdistro

### PR DESCRIPTION
**Pull request by mgonzs13**
# Please add YASMIN to be indexed in the rosdistro

## Package name:

`yasmin`

## Package Upstream Source:

https://github.com/uleroboticsgroup/yasmin.git

## Purpose of using this:

`YASMIN` is a project focused on implementing robot behaviors using Finite State Machines (FSM). It is available for ROS 2, Python and C++. `YASMIN` includes default states for ROS 2 subscribers, clients and actions and uses blackboards to share data between states and state machines. Furthermore, state machines can be canceled, which means canceling the current executing state. Finally, a web viewer is included, which allows monitoring of the execution of the state machines.

# Checks
 - [x] All packages have a declared license in the package.xml
 - [x] This repository has a LICENSE file
 - [x] This package is expected to build on the submitted rosdistro

### CODE REVIEW

File: iron/distribution.yaml [hunk 1]
```diff
@@ -8947,6 +8947,16 @@ repositories:
...
+  yasmin:
+    doc:
+      type: git
+      url: https://github.com/uleroboticsgroup/yasmin.git
+      version: main
+    source:
+      type: git
+      url: https://github.com/uleroboticsgroup/yasmin.git
+      version: main
+    status: developed
```
Comment (clalancette) [inline iron/distribution.yaml:8959]: Please remove the stanza for `iron`; it is in the last hold sync, so there is no opportunity to add new packages to Iron anymore.

**Review by clalancette — DISMISSED** [review]

File: iron/distribution.yaml (hunk 1 above)
Comment (mgonzs13) [reply iron/distribution.yaml]: Done, sorry for including it.

**Comment by mjcarroll** [comment]
[Holding for Jazzy sync](https://discourse.ros.org/t/preparing-for-jazzy-sync-2024-11-21/40682)

**Comment by reinzor** [comment]
@mgonzs13 , I think we also want to add a release with a release repository so that a specific version will be shipped using `apt`: https://docs.ros.org/en/jazzy/How-To-Guides/Releasing/First-Time-Release.html

**Comment by mgonzs13** [comment]
@clalancette I've removed the package from Iron. Is there something else to do?

**Review by audrow — APPROVED** [review]

**Pull request merged**